src/libAtomVM/opcodesswitch.h

```diff
@@ -1612,6 +1619,13 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
 
                 #ifdef IMPL_EXECUTE_LOOP
                     ctx->flags &= ~WaitingTimeoutExpired;
+
+                    struct ListHead *item;
+                    struct ListHead *tmp;
+                    MUTABLE_LIST_FOR_EACH(item, tmp, &ctx->save_queue) {
```

Comment:
I found that if I change this to be a non-mutable iteration, the loop will never terminate, because the `list_prepend` operation _does_ mutate the item.  Hence the test in the macro (`item != (head)`) will never be true.

Reply:
You're right, I didn't think about that they are different queues but the same list items are reused.